src/app/components/user/UserWorkspaces/UserWorkspacesComponent.js

```diff
@@ -0,0 +1,243 @@
+      id="switchTeams.error"
+      defaultMessage="Sorry, an error occurred while updating the workspace. Please try again and contact {supportEmail} if the condition persists."
+      description="Error message with instructions on how to contact support"
+      values={{ supportEmail: stringHelper('SUPPORT_EMAIL') }}
+    />;
+    setFlashMessage(errorMessage, 'error');
+  };
+
+  const onSuccess = (team) => {
+    const path = `/${team.slug}/all-items`;
+    browserHistory.push(path);
+  };
+
+  const setCurrentTeam = (team) => {
+    commitMutation(Relay.Store, {
+      mutation: updateUserMutation,
+      variables: {
+        input: {
+          current_team_id: team.dbid,
+        },
+      },
+      onCompleted: ({ error }) => {
+        if (error) {
+          onFailure(error);
+        } else {
+          onSuccess(team);
+        }
+      },
+      onError: onFailure,
+    });
+  };
+
+  const handleCancelCreateTeam = () => {
+    setShowCreateTeamDialog(false);
+  };
+
+  return (
+    <>
+      <SettingsHeader
+        title={
+          <FormattedMessage
+            id="userSettings.workspacesTitle"
+            defaultMessage="Workspaces [{workspacesCount}]"
+            description="Title for user settings area for user workspaces list they are a memeber of"
+            values={{ workspacesCount: numberOfTeams || 0 }}
+          />
+        }
+        actionButton={
+          <ButtonMain
+            theme="brand"
+            size="default"
+            variant="contained"
+            onClick={() => setShowCreateTeamDialog(true)}
+            label={
+              <FormattedMessage id="switchTeams.newTeamLink" defaultMessage="Create" description="Button label for initiating creating a new team workspace" />
+            }
+          />
+        }
+      />
+      { totalCount > pageSize && // only display paginator if there are more than pageSize worth of workspaces overall in the database
+        <div className={cx('paginator', workspaceStyles['workspaces-wrapper'])}>
+          <Tooltip
+            arrow
+            title={
+              <FormattedMessage id="search.previousPage" defaultMessage="Previous page" description="Pagination button to go to previous page" />
+            }
+          >
+            <span>
+              <ButtonMain
+                disabled={isPaginationLoading || cursor - pageSize < 0}
+                variant="text"
+                theme="brand"
+                size="default"
+                onClick={() => {
+                  if (cursor - pageSize >= 0) {
+                    setCursor(cursor - pageSize);
+                  }
+                }}
+                iconCenter={<PrevIcon />}
+              />
+            </span>
+          </Tooltip>
+          <span className={cx('typography-button', styles['workspace-header-count'])}>
+            <FormattedMessage
+              id="searchResults.itemsCount"
```

Comment:
Please fix the namespace to match the component name.

Reply:
I'm pair-reviewing this with Sarah and asked to keep existing string ids to avoid generating retranslations.